### PR TITLE
Feature/server hashing

### PR DESCRIPTION
**Pull request by AlbertLapTu**
Please review my code. Added the following:

- Hash/Salt/ functions to send hashed user data to the database
- Standard form fields with placeholder password strength bar.

### CODE REVIEW

File: __tests__/LoginForm.test.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  });
+
+  it('Should require alphanumeric passwords', () => {
+
+  });
+
+  it('Should validate passwords')
+
+  it('Should allow user\'s to register, () => {
```
Comment (fdistor) [inline __tests__/LoginForm.test.js:15]: Missing ' after register

File: client/src/components/loginForm.jsx [hunk 2]
```diff
@@ -1,22 +1,56 @@
...
+      username: '',
+      password: '',
+      strength: '',
+      color: ''
+    };
+  }
+
+  handleInput = e => {
+    this.setState({
+      [`${e.target.name}`]: e.target.value
```
Comment (fdistor) [inline client/src/components/loginForm.jsx:16]: I personally would deconstruct e.target first then assign the variables.

Comment (fdistor) [reply client/src/components/loginForm.jsx:16]: Also I don't think you need to put it between the key between brackets.

File: client/src/components/loginForm.jsx [hunk 3]
```diff
@@ -1,22 +1,56 @@
...
+  };
+
+  render() {
+    return (
+      <form>
+        <div id='username-container'>
+          <label htmlFor='username' id='username'>
+            <span>Username</span>
+            <input
+              onChange={this.handleInput}
```
Comment (fdistor) [inline client/src/components/loginForm.jsx:27]: I might be wrong, but doesn't onChange this.handleInput need to be bound?

File: client/src/components/loginForm.jsx [hunk 4]
```diff
@@ -1,22 +1,56 @@
...
+              type='text'
+              name='username'
+              placeholder='Enter username...'
+            />
+          </label>
+        </div>
+        <div id='password-container'>
+          <label htmlFor='password' id='password'>
```
Comment (fdistor) [inline client/src/components/loginForm.jsx:35]: What's the tag "htmlFor" used for? Just curious.

**Review by fdistor — CHANGES_REQUESTED** [review]
Good spacing and easy to read, but you should use double quotes when writing HTML elements. Also write your tests out lol.

File: client/src/components/loginForm.jsx (hunk 3 above)
Comment (AlbertLapTu) [reply client/src/components/loginForm.jsx:27]: Using the @babel/plugin-proposal-class-properties plugin allows for ES6 arrow functions to be used as methods on a class. 

The class properties plugin will auto bind this to the current class, so there isn't a need for binding within the constructor.

File: client/src/components/loginForm.jsx (hunk 4 above)
Comment (AlbertLapTu) [reply client/src/components/loginForm.jsx:35]: HTML markup uses a "for" attribute specifies which part of the form a label is bound to. Essentially just provides additional markup for the browser to recognize what to do with the label when it's time to be displayed. 

"htmlFor" is just React's way of naming it as it's own native attribute. Think: "class in HTML vs className in React).

**Comment by AlbertLapTu** [comment]
Closing pull request as it's not totally complete. I've found an npm package that is a de facto standard and will refactor to utilize that.

**Pull request closed without merging**